docs/data/material/customization/css-layers/css-layers.md

```diff
@@ -0,0 +1,250 @@
+
+```tsx title="pages/_app.tsx"
+import { AppCacheProvider } from '@mui/material-nextjs/v15-pagesRouter';
+import GlobalStyles from '@mui/material/GlobalStyles';
+
+export default function MyApp(props: AppProps) {
+  const { Component, pageProps } = props;
+  return (
+    <AppCacheProvider {...props}>
+      <GlobalStyles styles="@layer theme, base, mui, components, utilities;" />
+      <Component {...pageProps} />
+    </AppCacheProvider>
+  );
+}
+```
+
+### Vite or any other SPA
+
+Make the following changes in `src/main.tsx`:
+
+1. Pass the `enableCssLayer` prop to the `StyledEngineProvider` component.
+2. Configure the layer order with the `GlobalStyles` component to work with Tailwind CSS v4.
+
+```tsx title="main.tsx"
+import { StyledEngineProvider } from '@mui/material/styles';
+import GlobalStyles from '@mui/material/GlobalStyles';
+
+ReactDOM.createRoot(document.getElementById('root')!).render(
+  <React.StrictMode>
+    <StyledEngineProvider enableCssLayer>
+      <GlobalStyles styles="@layer theme, base, mui, components, utilities;" />
+      {/* Your app */}
+    </StyledEngineProvider>
+  </React.StrictMode>,
+);
+```
+
+## Multiple layers
+
+On top of the [single layer](#single-layer), you can split styles into multiple layers to better organize them within Material UI.
+This makes theming and overriding styles through the `sx` prop easier.
+
+Follow the steps from the [previous section](#single-layer) to enable the CSS layer feature.
+Then, create a new file and export the component that wraps the `ThemeProvider` from Material UI.
+Finally, pass the `experimental_modularCssLayers: true` option to the `createTheme` function:
+
+```tsx title="src/theme.tsx"
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+const theme = createTheme({
+  experimental_modularCssLayers: true,
+});
+
+export default function AppTheme({ children }: { children: ReactNode }) {
+  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
+}
+```
+
+{{"demo": "CssLayersInput.js"}}
+
+When this feature is enabled, Material UI generates these layers:
+
+- `@layer mui.global`: The global styles from `GlobalStyles` and `CssBaseline` components.
+- `@layer mui.default`: The base styles for all Material UI components.
+- `@layer mui.theme`: The theme styles for all Material UI components.
+- `@layer mui.custom`: The custom styles for non-Material UI styled components.
+- `@layer mui.sx`: The styles from the `sx` prop.
+
+If you want to integrate with other styling solutions, such as Tailwind CSS v4, you can specify the layer order as a value to the `experimental_modularCssLayers` field, Material UI will replace the `mui` identifier with the correct order:
+
+```diff title="src/theme.tsx"
+ const theme = createTheme({
+-  experimental_modularCssLayers: true,
++  experimental_modularCssLayers: '@layer theme, base, mui, components, utilities;',
```

Comment:
@mapache-salvaje based on the docs, feel free to suggest a different name for enabling this feature 😅.